sky/clouds/cloud.py

```diff
@@ -309,6 +309,21 @@ def get_feasible_launchable_resources(self, resources):
     def _get_feasible_launchable_resources(self, resources):
         raise NotImplementedError
 
+    def get_available_reservation_resources(
+        self,
+        instance_type: str,
+        region: str,
+        zone: Optional[str],
+        specific_reservations: List[str],
+    ) -> int:
+        """"
+        Returns the number of available reservation resources for the given
+        instance type in the given region/zone.
+        Default implementation returns 0 for non-implemented clouds.
+        """
+        _ = instance_type, region, zone, specific_reservations
```

Comment:
nit: we use `del` for unused parameters. 
https://google.github.io/styleguide/pyguide.html#214-decision

Reply:
Done